### PR TITLE
Allow toggling of visible scrollbar indicators

### PR DESCRIPTION
**Pull request by sitefinitysteve**
Implements #4522

Okay, so I wasn't able to get the tests running here locally... kept crashing.  I took the source and put it in a hello world to test with.

### CODE REVIEW

**Comment by vakrilov** [comment]
Hey @sitefinitysteve - thanks for the effort.
We have recently published a [contributing guideline](https://github.com/NativeScript/NativeScript/blob/master/CONTRIBUTING.md). You can give it a try - it explains how to run and write unit tests. 
That said - unit tests are really essential for maintaining the quality of the project, so we would appreciate if you can write test to test this feature. There are bunch of other [scroll-view related tests here](https://github.com/NativeScript/NativeScript/blob/master/tests/app/ui/scroll-view/scroll-view-tests.ts).

**Comment by sitefinitysteve** [comment]
I know about the new guides, saw your tweet and followed the instructions :)

But running the app and tests all just fail... Not like the tests fail, but
fail to run.

Maybe it's that i need 3.1.1 loaded and I'm only on 3.0 (can't risk an
update breaking things, important app about to ship).

What if i write the test as i assume it would pass... Would the ci here
pick it up and run it?

**Comment by vakrilov** [comment]
run ci

**Comment by sitefinitysteve** [comment]
Okay I added a test, I can't get it to run on iOS (per the [docs](https://github.com/NativeScript/NativeScript/blob/master/DevelopmentWorkflow.md#running-unit-tests)), but seems to pass on Android (I mean fails on someones grid test though...

**Comment by vakrilov** [comment]
run ci

**Comment by sitefinitysteve** [comment]
I can't see the unit test build failed results... internal server or has to be published back out?

**Comment by vakrilov** [comment]
Hi,
The unit test app failed due to some changes in the CI we are currently making internally. 
I will trigger a new build as soon as the changes are done and will notify you if there are failing tests.

Thanks for the unit tests.
One other thing would you mind renaming the new property:
* `scrollBar` -> `scrollBarVisible`
* `scrollBarProperty` -> `scrollBarVisibleProperty`

**Comment by sitefinitysteve** [comment]
Sure, one step further, scrollBarIndicatorVisible?

**Comment by vakrilov** [comment]
That's even better.

**Comment by vakrilov** [comment]
run ci

**Comment by vakrilov** [comment]
run ci

File: tns-core-modules/ui/scroll-view/scroll-view.ios.ts [hunk 1]
```diff
@@ -146,7 +161,7 @@ export class ScrollView extends ScrollViewBase {
     }
 
     public _onOrientationChanged() {
-        // NOOP
+        this.updateScrollBarVisibility(this.scrollBar);
```
Comment (vakrilov) [inline tns-core-modules/ui/scroll-view/scroll-view.ios.ts]: Getting TypeScript transpile error here:
`error TS2339: Property 'scrollBar' does not exist on type 'ScrollView'.`
Should be `this.scrollBarIndicatorVisible`

File: tns-core-modules/ui/scroll-view/scroll-view-common.ts [hunk 2]
```diff
@@ -77,6 +77,10 @@ export abstract class ScrollViewBase extends ContentView implements ScrollViewDe
         return 0;
     }
 
+    get scrollBarIndicatorVisible(): boolean {
```
Comment (vakrilov) [inline tns-core-modules/ui/scroll-view/scroll-view-common.ts]: Can you make this just:
`public scrollBarIndicatorVisible: boolean;`
This property will be re-defined when registering the property, so returning a value can confuse someone that this is the actual implementation.

Comment (vakrilov) [inline tns-core-modules/ui/scroll-view/scroll-view-common.ts]: It is similar to the `orientation` property in the beginning of the class - you can also move it there

File: tns-core-modules/ui/scroll-view/scroll-view-common.ts [hunk 3]
```diff
@@ -94,4 +98,10 @@ export const orientationProperty = new Property<ScrollViewBase, Orientation>({
     },
     valueConverter: converter
 });
-orientationProperty.register(ScrollViewBase);
\ No newline at end of file
+orientationProperty.register(ScrollViewBase);
+
+export const scrollBarIndicatorVisibleProperty = new Property<ScrollViewBase, boolean>({
+    name: "scrollBar", defaultValue: true,
```
Comment (vakrilov) [inline tns-core-modules/ui/scroll-view/scroll-view-common.ts]: Change the name -> `"scrollBarIndicatorVisible"`

File: tests/app/ui/scroll-view/scroll-view-tests.ts [hunk 4]
```diff
@@ -254,6 +254,48 @@ class ScrollLayoutTest extends testModule.UITest<scrollViewModule.ScrollView> {
...
         TKUnit.assertEqual(scrollX, layoutHelper.dp(100));
     }
+
+    public test_scrollView_horizontal_can_set_indicator_state() {
+        this.testView.orientation = "horizontal";
+        this.testView.scrollBarIndicatorVisible = true;
+        this.waitUntilTestElementLayoutIsValid();
+
+        if (app.ios) {
+            TKUnit.assertEqual(this.testView.ios.showsHorizontalScrollIndicator, true);
```
Comment (vakrilov) [inline tests/app/ui/scroll-view/scroll-view-tests.ts]: Use `nativeView` instead of `ios/android`

**Comment by vakrilov** [comment]
Hi @sitefinitysteve 
I've requested some changes. After the rename the project isn't building. 
It is a good idea to build the project and make sure all tests are green when pushing in the PR branch.

If you are still having problems running the tests on both platforms - just ping me on slack to get this sorted. Being able to write and run the unit test is an essential prerequisite to a successful PR.

**Comment by sitefinitysteve** [comment]
Sure, i swear to god on Android all the tests passed, i did run it. :/

**Comment by vakrilov** [comment]
run ci

**Pull request merged**